check_binary.sh

```diff
@@ -34,83 +35,104 @@ fi
 ###############################################################################
 # Check GCC ABI
 ###############################################################################
-echo "Checking that the gcc ABI is what we expect"
-if [[ "$(uname)" != 'Darwin' ]]; then
-  function is_expected() {
-    # This commented out logic is what you'd expect if 'devtoolset7' actually
-    # built with the new GCC ABI, but it doesn't; it always builds with ABI=0.
-    # When a compiler is added that does build with new ABI, then replace
-    # devtoolset7 (and the DESIRED_DEVTOOLSET variable) with your new compiler
-    #if [[ "$DESIRED_DEVTOOLSET" == 'devtoolset7' ]]; then
-    #  if [[ "$1" -gt 0 || "$1" == "ON" ]]; then
-    #    echo 1
-    #  fi
-    #else
-    #  if [[ -z "$1" || "$1" == 0 || "$1" == "OFF" ]]; then
-    #    echo 1
-    #  fi
-    #fi
-    if [[ -z "$1" || "$1" == 0 || "$1" == "OFF" ]]; then
-      echo 1
+
+# NOTE [ Building libtorch with old vs. new gcc ABI ]
```

Comment:
Any substantive changes here?

Reply:
There is no substantive change here. The only main change is that we are only checking for cxx11 / pre-cxx11 symbols in our own namespaces (i.e. `c10` / `at` / `caffe2` / `torch`), since checking for those symbols in all namespaces doesn't give us reliable results:
```
  # NOTE: Checking the above symbols in all namespaces doesn't work, because
  # devtoolset7 always produces some cxx11 symbols even if we build with old ABI,
  # and CuDNN always has pre-cxx11 symbols even if we build with new ABI using gcc 5.4.
  # Instead, we *only* check the above symbols in the following namespaces:
```